storage/azure/src/server/wrappers/BlobServiceClientWrapper.ts

```diff
@@ -41,10 +49,70 @@ export class BlobServiceClientWrapper {
       .getBlockBlobClient(buildBlobName(reference));
   }
 
-  public listContainers(): PagedAsyncIterableIterator<
-    ContainerItem,
-    ServiceListContainersSegmentResponse
-  > {
-    return this._client.listContainers();
+  public async getDirectoriesNextPage(options: {
+    maxPageSize: number;
+    continuationToken?: string;
+  }): Promise<EntityCollectionPage<BaseDirectory>> {
+    const iterator = this._client.listContainers().byPage({
+      maxPageSize: options.maxPageSize,
+      continuationToken: options.continuationToken,
+    });
+
+    const response = (await iterator.next()).value;
+    const directories = response.containerItems.map(
+      (directory: ContainerItem) =>
+        ({ baseDirectory: directory.name } as BaseDirectory)
+    );
+    const ret: EntityCollectionPage<BaseDirectory> = {
+      entities: directories,
+      next:
+        response.continuationToken == ""
+          ? undefined
+          : () =>
+              this.getDirectoriesNextPage({
+                maxPageSize: options.maxPageSize,
+                continuationToken: response.continuationToken,
+              }),
+    };
+    return ret;
```

Comment:
We're usually avoiding abbreviations. Especially here this doesn't really save any horizontal space, compared to something like `result`, `page` or `returnValue`. We're usually only using abbreviations for something where abbreviation is at least as common or even more common than the word/phrase itself (Db, Url)